### PR TITLE
feat: Add Rows-CSV format to SFM export

### LINKED ISSUE #74: feat: Request to export CSV
@sdysart is supporting some users who are converting wordlists from SA into Cog.
Currently, Cog is expecting .csv files in the format:
row 1 - glosses separated by commas
row 2 - secondary field (IPA?)

### Workaround
@sdysart starts with SA exporting to Fieldworks SFM.
Then, he's applying a regex in Notepad++ to create CSV (data in column 1 and column 2).
Then, he's applying a Google sheet transpose to get the data into 2 rows.

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #74 

To support users exporting from SA to Cog, this inserts an option "Rows Comma-Separated Values" to the SFM export dialog.
This option uses the `,` character to separate the entries instead of spaces ` `.

### Limitations
* If a field contains a comma, the user will need to surround it in `"` quotes in order to prevent the field from getting split.
* The resulting .csv file will need to be converted to UTF-8 BOM (e.g. in Notepad++)

### CODE REVIEW

**Comment by darcywong00** [comment]
Some initial comments from @sdysart:

> I took a look at the csv feature you added to SA.  
The export as CVS option will really help.
I did notice that in the export, the phonetic row was off by one towards the end, that was because they had inserted a comma in one of the phonetic fields
>
>I guess they had not decided how to proceed.  I suppose that we can assume that the users will clean up the data prior to export, but it should be noted in the instructions somewhere
that comma's can't be in the exported fields if exporting as csv.  It would be easily to fix. but it should be noted.
>
>The export still exports as SFM, not csv.  I had to rename it in File Explorer, which is easy enough to do.
Also, Excel doesn't seem to like the way SA exports IPA.  It goes weird.  When I went through the process of editing the sfm in Notepad++ Tim E. pointed out that I had to mark the encoding as UTF-8 BOM in Notepad++ in order for Excel to pick it up.
LibreOffice Calc asks me to pick the encoding when opening the csv file.

**Comment by megahirt** [comment]
We should be able to escape commas in the standard CSV way.  I'm thinking
that just means that each field has quotes around it.  Best way to test is
to take the offending CSV that @sdysart mentions and put quotes around the
field in the line where the data has a comma.  See if it then imports
correctly.

**Comment by darcywong00** [comment]
> Best way to test is
to take the offending CSV that [@sdysart](https://github.com/sdysart) mentions and put quotes around the
field in the line where the data has a comma.  See if it then imports
correctly.

Yeah, we can try that next time. Though we'd also have to escape quotes within the string. For Excel, it might be `""`?

File: Src/SA/DlgExportSFM.cpp [hunk 1]
```diff
@@ -255,17 +258,19 @@ void CDlgExportSFM::OnOK() {
         m_bLanguage = m_bDialect = m_bSpeaker = m_bGender = m_bEthnologue = m_bFamily = m_bRegion = m_bNotebookRef =
                 m_bTranscriber = m_bComments = m_bCountry = TRUE;
 
+    m_szSeparator = (m_nExportFormat != 2) ? L" " : L",";
```
Comment (kgorham) [inline Src/SA/DlgExportSFM.cpp:261]: Is there any value in having m_szSeparator as a class member.  I'm thinking it should just be local scope.
I see why you did it, but it bothers me a little that the Export* methods are protected, meaning that they can be called by a derived class and end up with an un-initialized m_szSeparator.  So either the Exports* could be private, or pass the separator in as a parameter.

Comment (darcywong00) [reply Src/SA/DlgExportSFM.cpp]: Yep. I've updated to pass a `separator` parameter to the applicable Exports*, with a default separator `L" "`.

**Comment by darcywong00** [comment]
> We should be able to escape commas in the standard CSV way.  I'm thinking
that just means that each field has quotes around it.  Best way to test is
to take the offending CSV that [@sdysart](https://github.com/sdysart) mentions and put quotes around the
field in the line where the data has a comma.  See if it then imports
correctly.

@sdysart tested the latest build. We found if he surrounds a field in `"` quotes, the CSV doesn't split the line with a comma.

**Review by megahirt — APPROVED** [review]

**Pull request merged**